### ISSUE TITLE
Document Deploying Private Helm Repository Hosted on Github

### ISSUE THREAD

**Issue by andrewsingley-glv** [fluxcd/website#2005]
When I deployed a private helm repository using flux, I first ran the following command:

```flux help create source helm```

... and I was given the following example:

```
  # Create a source for an HTTPS Helm repository using basic authentication
  flux create source helm podinfo \
    --url=https://stefanprodan.github.io/podinfo \
    --username=username \
    --password=password
```
 
The url in the address above is a Github Pages address. This led me down a rabbit hole. After lots of research, I realized that private Github Pages does not support basic auth. However, Github does host a raw endpoint that does, `https://raw.githubusercontent.com`. This article explains how this endpoint can be used for private Helm repositories, [Stuart Blackler](https://im5tu.io/article/2022/01/creating-a-private-helm-repository-using-github-pages-enterprise/ ) .

TLDR: The URL needs to be in the following format: `https://raw.githubusercontent.com/<organization>/<repo-path>/<branch>/`

The URL above, including the trailing slash should be passed as the parameter to the above command.

Why does FluxCD need to document this? Shouldn't developers refer to Github to document Github endpoints? In an ideal world, yes, but Github has already declined to do so. See this closed issue: https://github.com/github/docs/issues/8031#issuecomment-1709184264

I suggest replacing the example in the CLI at minimum, like the modified version, below. Discussing the format of the URL in the HelmRepository docs on the FluxCD website would be even better.

```
  # Create a source for an HTTPS Helm repository using basic authentication
  flux create source helm podinfo \
    --url=https://raw.githubusercontent.com/stefanprodan/podinfo/main/ \
    --username=username \
    --password=password
```

Also, I tested Stuart's solution and it does work.

**Comment by stefanprodan** [comment]
With Helm OCI fully supported in GitHub Container Registry, I really hope Flux users have moved away from HTTP Helm repos.

**Comment by andrewsingley-glv** [comment]
Storing helm charts in OCI registries is really cool, but it is still fairly new. 

My use case was extremely simple. I just had one value to template in one file, so I just wanted to make the simplest Helm repository possible. If there was a way to embed the helm template directly into the HelmChart resource of the source.toolkit.fluxcd.io/v1 API, that would have been ideal. Then my helm resources could coreside with the rest of my code pulled in from a GitRepository resource. This is totally unrelated to documentation, but it would resolve my original problem that led to my low-effort helm struggle.

**Comment by andrewsingley-glv** [comment]
@stefanprodan Helm has not fully committed to OCI packages because their [chart-releaser](https://github.com/helm/chart-releaser/issues/183) action for GitHub still does not support OCI at this time. This is heavily used by the Helm Community for generating packages, the tarballs that get pushed to clients. Helm 3.8.0 appears to be the first CLI version to support OCI and was only pushed out December '23, making it barely six months now. I think it's reasonable to support HTTP helm repositories for just a little longer. Thank you for your time.

**Comment by andrewsingley-glv** [comment]
@stefanprodan While I was looking through that issue, I saw someone mention FluxCD and that was one link away from a GitHub Action you approved in the PR, pushing to OCI in GHCR. https://github.com/fluxcd-community/helm-charts/blob/main/.github/workflows/release.yml

**Comment by stefanprodan** [comment]
Flux can build charts from Git, see the docs here: 
- https://fluxcd.io/flux/guides/helmreleases/#git-repository
- https://fluxcd.io/flux/components/source/helmcharts/#reconcile-strategy

**Comment by andrewsingley-glv** [comment]
Thank you. I think this is sufficient. I wanted to see the example URL of a private helm git repository updated so developers didn't expect a github page to accept basic authentication. The fact that Github hasn't publicly documented the  https://raw.githubusercontent.com, which does accept basic auth, makes me hesitate to put it in the documentation.
Again, my concern was just with the example URL in the CLI for private HTTP helm repositories.

**Comment by andrewsingley-glv** [comment]
I will go ahead and close this issue. Thank you for your time and consideration.
